include/silo/storage/column/indexed_string_column.h

```diff
@@ -0,0 +1,72 @@
+#ifndef SILO_INDEXED_STRING_COLUMN_H
+#define SILO_INDEXED_STRING_COLUMN_H
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <boost/archive/binary_iarchive.hpp>
+#include <boost/archive/binary_oarchive.hpp>
+#include <roaring/roaring.hh>
+
+#include "silo/common/bidirectional_map.h"
+#include "silo/storage/column/column.h"
+
+namespace boost::serialization {
+struct access;
+}
+
+namespace silo::storage::column {
+
+class IndexedStringColumnPartition : public Column {
+   friend class boost::serialization::access;
+
+  private:
+   template <class Archive>
+   [[maybe_unused]] void serialize(Archive& archive, const unsigned int /* version */) {
+      // clang-format off
+      archive & value_ids;
+      archive & indexed_values;
+      // clang-format on
+   }
+
+   std::vector<Idx> value_ids;
+   std::unordered_map<Idx, roaring::Roaring> indexed_values;
+   common::BidirectionalMap<std::string>& lookup;
+
+  public:
+   explicit IndexedStringColumnPartition(common::BidirectionalMap<std::string>& lookup);
+
+   [[nodiscard]] roaring::Roaring filter(const std::string& value) const;
+
+   void insert(const std::string& value);
+
+   const std::vector<silo::Idx>& getValues() const;
+};
+
+class IndexedStringColumn : public Column {
+   friend class boost::serialization::access;
```

Comment:
Why do we actually need that friend class here? We didn't really test it yet, but at least it compiled without it where I implemented the serialize thingy.

Reply:
I thought it is more consistent style to have the serialization as a private member instead of static, where we then need the friend class

Reply:
Ok, valid :+1: